src/huggingface_hub/hf_api.py

```diff
@@ -1868,6 +1875,109 @@ def upload_file(
         d = r.json()
         return d["url"]
 
+    def commit_files(
+        self,
+        *,
+        repo_id: str,
+        files: Iterable[FileUploadInput],
+        commit_summary: str,
+        commit_description: Optional[str] = None,
+        token: Optional[str] = None,
+        repo_type: Optional[str] = None,
+        revision: Optional[str] = None,
+    ):
```

Comment:
shouldn't we do a more general `create_commit` which lets one delete and add (or update) files in the same call?

Reply:
like `file_changes: FileChanges` with `FileChanges` is `{ additions: Obj[]; deletions: Obj[] }`

or `files: Iterable[FileUploadInput] = []` AND `deleted_files: Iterable[str] = []` which throws if both arrays are empty?